### PR TITLE
Add variable to query debug mode

### LINKED ISSUE #180: Debug mode can be queried from variable
See this [thread](https://github.com/gradle/gradle-enterprise-build-validation-scripts/pull/174#discussion_r1008814724).

### PR DESCRIPTION
**Pull request by erichaagdev**
## ❄️ Winter Update PR Navigator ❄️

1. #261
2. #262
3. #263
4. #264
5. #265
6. ⛄ #266
7. #267
8. #272
9. #274
10. #283

⛄ = you are here

## Summary

This PR adds a global `debug_mode` variable that can queried throughout the scripts. Previously the variable generated by Argbash was being used. 

Closes #180

### CODE REVIEW

**Review by runningcode — APPROVED** [review]
nice!

File: components/scripts/gradle/04-validate-remote-build-caching-ci-ci.sh [hunk 1]
```diff
@@ -327,7 +327,7 @@ generate_command_to_repeat_experiment() {
     cmd+=("-f")
   fi
 
-  if [[ "${_arg_debug}" == "on" ]]; then
+  if [[ "${debug_mode}" == "on" ]]; then
```
Comment (runningcode) [inline components/scripts/gradle/04-validate-remote-build-caching-ci-ci.sh:330]: why the name change here?

Comment (erichaagdev) [reply components/scripts/gradle/04-validate-remote-build-caching-ci-ci.sh:330]: `_arg_x` variables are generated by Argbash which is what we use for parsing command line parameters, displaying help text, etc. One of the first things we do in the scripts is call this [`process_arguments`](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/18840cc8647dca8976535e7784919bcb0041bb2c/components/scripts/lib/config.sh#L4) function which maps the Argbash generated variables to our own internal variables. This way we have more control over their naming.

For some reason `_arg_debug` was the only we didn't map and called it directly throughout the codebase. See [this thread](https://github.com/gradle/gradle-enterprise-build-validation-scripts/pull/174#discussion_r1008193504) for additional context.

Comment (runningcode) [reply components/scripts/gradle/04-validate-remote-build-caching-ci-ci.sh:330]: Thanks for the context!

**Review by jthurne — APPROVED** [review]

**Review by etiennestuder — APPROVED** [review]

**Pull request merged**